### PR TITLE
feat: upgrade to openresty-1.25.3.1

### PR DESCRIPTION
**Pull request by zll600**

### CODE REVIEW

File: patch/1.25.3.1/ngx_lua-shared_shdict.patch [hunk 1]
```diff
@@ -0,0 +1,2965 @@
...
+ 
+     ngx_destroy_pool(c->pool);
+     return NGX_OK;
+@@ -314,12 +315,12 @@ failed:
+ 
+ ngx_int_t
+ ngx_http_lua_init_worker_by_inline(ngx_log_t *log,
+-    ngx_http_lua_main_conf_t *lmcf, lua_State *L)
++    ngx_str_t init_worker_src, lua_State *L)
+ {
```
Comment (zll600) [inline patch/1.25.3.1/ngx_lua-shared_shdict.patch:582]: The implementation has been change in ngx_lua-0.10.26. So the patch will fail to apply

File: patch/1.25.3.1/lua-resty-core-shared_shdict.patch [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+-long ngx_http_lua_ffi_shdict_get_ttl(void *zone,
++long ngx_meta_lua_ffi_shdict_get_ttl(void *zone,
+     const unsigned char *key, size_t key_len);
+ 
+-int ngx_http_lua_ffi_shdict_set_expire(void *zone,
++int ngx_meta_lua_ffi_shdict_set_expire(void *zone,
+     const unsigned char *key, size_t key_len, long exptime);
+ 
+-size_t ngx_http_lua_ffi_shdict_capacity(void *zone);
+-
```
Comment (zll600) [inline patch/1.25.3.1/lua-resty-core-shared_shdict.patch:50]: The implementation has been change in ngx_lua-0.1.28. So the patch will fail to apply

**Review by zll600 — COMMENTED** [review]
There are some problems when I update the patch files.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -39,7 +40,7 @@ jobs:
 
       - name: Install
         run: |
-          wget https://raw.githubusercontent.com/api7/apisix-build-tools/master/build-apisix-base.sh
+          wget https://raw.githubusercontent.com/zll600/apisix-build-tools/upgrade_openresty-1.25.3.1/build-apisix-runtime.sh
```
Comment (zll600) [inline .github/workflows/ci.yml:43]: I want to test the patch file by ci. So I change the download link.

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -29,17 +28,17 @@ jobs:
...
       - name: Before install
         run: |
           sudo cpanm --notest Test::Nginx > build.log 2>&1 || (cat build.log && exit 1)
-          git clone https://github.com/iresty/test-nginx.git test-nginx
+          git clone https://github.com/openresty/test-nginx.git test-nginx
 
       - name: Install
         run: |
-          wget https://raw.githubusercontent.com/api7/apisix-build-tools/master/build-apisix-base.sh
+          wget https://raw.githubusercontent.com/zll600/apisix-build-tools/upgrade_openresty-1.25.3.1/build-apisix-base.sh
```
Comment (monkeyDluffy6017) [inline .github/workflows/ci.yml:41]: remember to revert this change

File: t/pipe.t [hunk 5]
```diff
@@ -210,6 +210,6 @@ exited
     }
 --- response_body
 ok
---- shutdown_error_log
+--- error_log
```
Comment (monkeyDluffy6017) [inline t/pipe.t:213]: Why do you change this?
The `--- shutdown_error_log` section is used for checking the error logs generated during nginx exit.
The `--- error_log` to test normal error logs
I think they are for different purpose

Comment (zll600) [reply t/pipe.t:213]: I found that log appears in `error_log`, but not in `shutdown_error_log`. also confirmed this by modifying the source code of test-nginx.

File: .github/workflows/ci.yml (hunk 4 above)
Comment (membphis) [reply .github/workflows/ci.yml:41]: this project can not depend `apisix-build-tools`

`apisix-build-tools` base on `apisix-nginx-module`

File: t/pipe.t (hunk 5 above)
Comment (membphis) [reply t/pipe.t:213]: we need to merge this PR first.

this change seems acceptable

@monkeyDluffy6017 if you have different though, you can create new issue or PR to fix it

**Review by membphis — APPROVED** [review]
add a todo issue: https://github.com/api7/apisix-nginx-module/pull/83/files#diff-b803fcb7f17ed9235f1e5cb1fcd2f5d3b2838429d4368ae4c57ce4436577f03fR41

we will revert this line after we merge https://github.com/api7/apisix-build-tools/pull/367
